src/test/java/com/example/demo/service/file/CustomFileReaderTest.java

```diff
@@ -0,0 +1,38 @@
+package com.example.demo.service.file;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+public class CustomFileReaderTest {
+
+    @Test
```

Comment:
I think you could add a couple tests for bad scenarios, e.g. trying to read a non-existent or empty file)

Reply:
agree